src/components/Datefilter/DateFilter.tsx

```diff
@@ -2,29 +2,41 @@ import { useRef, useState } from "react";
 import DatePicker from "react-datepicker";
 import "react-datepicker/dist/react-datepicker.css";
 import { Dropdown, Header } from "../Filter/style";
-import dateIcon from '../../assets/icons/date.svg';
+import dateIcon from "../../assets/icons/date.svg";
 import { useOnClickOutside } from "usehooks-ts";
+import { convertDateFromUi } from "../../utils/utils";
 
 export interface DateFilterProps {
-  name: string
-  onChangeValue: (startDate: Date | null, EndDate:Date | null) => void;
+  name: string;
+  onChangeValue: (startDate: any, EndDate: any) => void;
 }
 
 const DateFilter = (props: DateFilterProps) => {
-  const [startDate, setStartDate] = useState<Date | null>(null);
-  const [endDate, setEndDate] = useState<Date | null>(null);
+  const [startDate, setStartDate] = useState<any>('');
+  const [endDate, setEndDate] = useState<any>('');
   const ref = useRef(null);
   const [isActive, setIsActive] = useState(false);
   const handleClickOutside = () => {
     setIsActive(false);
   };
 
   useOnClickOutside(ref, handleClickOutside);
-  
+
   function handleDateChange(date: any) {
-      setStartDate(date[0]);
-      setEndDate(date[1]);
-      props.onChangeValue(startDate,endDate);
+    setStartDate(date[0]);
+    setEndDate(date[1]);
+    let start, end;
+    if(date[0]){
+      start = convertDateFromUi((date[0]).toString());
+    } else {
+      start ='';
+    }
+    if(date[1]){
+      end = convertDateFromUi((date[1]).toString());
+    }else {
+      end='';
+    }
```

Comment:
Although it is not wrong, we could instead of using if clauses and declaring variables you could shorten and make it readable like so
`const start = date[0] ? convertDateFromUi((date[0]).toString()) : '';
  const end = date[1] ? convertDateFromUi((date[1]).toString()) : '';
`
This way start and end will be dynamic depending on date[0] and date[1] and a bit more readable